tools/topology/topology2/platform/intel/sdw-jack-generic.conf

```diff
@@ -103,6 +108,11 @@ Object.Widget {
 			num_input_audio_formats 1
 			num_output_audio_formats 1
 			num_output_pins 1
+IncludeByKey.SDW_JACK_FORMAT {
+"[s*le]"	{
+					format		$SDW_JACK_FORMAT
+	}
+}
```

Comment:
I don't think it's a good idea. There is no reason to propagate restrictions on the Data Port WordLength register all the way to the gains

Reply:
@plbossart I dont think it propagated to the gain here. Still just the DAI copier

Reply:
right but still we have the valid_bit_depth so why do we need a format to represent what can be sent on the bus?

Reply:
Ahh so are you suggesting using the valid_bit_depth/container size in the output formats of the DAI copier? If it really only has support for 1 format, all of the output formats will have the same valid bit depth and container size and the kernel can infer the supported format from there?

Reply:
I am not sure if the kernel can do everything magically, my question is really whether there is a conceptual difference between valid_bit_depth and this new "format" suggestion.

it's also interesting because it's a limitation from the 3rd party chip that trickles into the topology, we've never had to deal with this before.

Reply:
I think using the valid_bit_depth can be used as a proxy for doing the sample format fixup. @bardliao what do you think?

Reply:
@plbossart @ranj063 valid_bit_depth  depends on hw_params. It will be the format that user play/capture. For example, it will be 16 when user play a s16_le wav.
The PR and the corresponding kernel PR is trying to specify the format of hw_params.

Reply:
> not following @bardliao, the hw_params depend on what the BE can support, not what applications request.

Right, but if the BE can support all formats, hw_params will be the format that applications request.
This comment let me think the issue is actually that ASoC doesn't change the hw_params if the format is not supported by the codec. For example, rt711-sdca supports `(SNDRV_PCM_FMTBIT_S16_LE | SNDRV_PCM_FMTBIT_S20_3LE | SNDRV_PCM_FMTBIT_S24_LE)`, but the format of hw_params is still s32_le if we play s32_le format file.

> 
> The point is that we have two concepts that seem completely identical, so why not just use one.

We set `valid_bit_depth` in the available formats and we can have multiple `valid_bit_depth` in a host widget.
And we can't use the `valid_bit_depth` in base format to specify the BE format in `sof_ipc4_pcm_dai_link_fixup` because base format will be decided in `sof_pcm_hw_params` -> `.ipc_prepare`.
And the `format` token is used to specify the format in IPC3. That's why I reuse the `format` token to specify the format in IPC4, too.